lab-judy/test/list-route-test.js

```diff
@@ -0,0 +1,296 @@
+        request.post(`${url}/api/list`)
+        .send(exampleList)
+        .end((err, res) => {
+          if (err) return done (err);
+          expect(res.status).to.equal(200);
+          expect(res.body.name).to.equal('judy');
+          this.tempList = res.body;
+          done();
+        });
+      });
+    });
+  });
+  //GET request 200
+  describe('testing GET request 200', function(){
+    describe('with valid id', function(){
+      before( done => {
+        exampleList.timestamp = new Date();
+        new List(exampleList).save()
+        .then( list => {
+          this.tempList = list;
+          done();
+        })
+        .catch(done);
+      });
+      after( done => {
+        delete exampleList.timestamp;
+        if(this.tempList){
+          List.remove({})
+          .then(() => done())
+          .catch(done);
+          return;
+        }
+        done();
+      });
+
+      it ('should return a list with status 200', done => {
+        debug('running it block for GET request with 200 status');
+        request.get(`${url}/api/list/${this.tempList._id}`)
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res.status).to.equal(200);
+          expect(res.body.name).to.equal('judy');
+          this.tempList = res.body;
+          done();
+        });
+      });
+    });
+  });
+
+  //PUT test 200
+  describe('testing PUT request 200', function(){
+    describe('with valid body', function(){
+      before( done => {
+        exampleList.timestamp = new Date();
+        new List(exampleList).save()
+        .then( list => {
+          this.tempList = list;
+          done();
+        })
+        .catch(done);
+      });
+      after( done => {
+        delete exampleList.timestamp;
+        if(this.tempList){
+          List.remove({})
+          .then(() => done())
+          .catch(done);
+          return;
+        }
+        done();
+      });
+      it('should update the data with 200 status', done => {
+        debug('running it block for PUT request with 200 status');
+        var updatedData = {name: 'Jen', age: 12};
+        request.put(`${url}/api/list/${this.tempList._id}`)
+        .send(updatedData)
+        .end((err, res) => {
+          if (err) return done (err);
+          expect(res.status).to.equal(200);
+          expect(res.body._id).to.equal(this.tempList._id.toString());
+          for (var key in updatedData){
+            expect(res.body[key]).to.equal(updatedData[key]);
+          }
+          this.tempList = res.body;
+          done();
+        });
+      });
+    });
+  });
+
+  //DELETE request 204
+  describe('testing DELETE request', function(){
+    describe('with valid ID', function(){
+      before( done => {
+        exampleList.timestamp = new Date();
+        new List(exampleList).save()
+        .then( list => {
+          this.tempList = list;
+          done();
+        })
+        .catch(done);
+      });
+      after( done => {
```

Comment:
You shouldn't need this after block if your delete test is working as it should.